contracts/sovereignChains/AgglayerBridgeL2FromEtrog.sol

```diff
@@ -0,0 +1,63 @@
+pragma solidity 0.8.28;
+
+import "./AgglayerBridgeL2.sol";
+
+// Contract created to perform the upgrade from the Etrog version to the AgglayerBridgeL2 version.
+contract AgglayerBridgeL2FromEtrog is AgglayerBridgeL2 {
+    /**
+     * @notice Override the function to prevent the contract from being initialized with this initializer
+     */
+    function initialize(
+        uint32, // _networkID
+        address, // _gasTokenAddress
+        uint32, // _gasTokenNetwork
+        IBaseLegacyAgglayerGER, // _globalExitRootManager
+        address, // _polygonRollupManager
+        bytes memory, // _gasTokenMetadata
+        address, // _bridgeManager
+        address, // _sovereignWETHAddress
+        bool, // _sovereignWETHAddressIsNotMintable
+        address, // _emergencyBridgePauser
+        address, // _emergencyBridgeUnpauser
+        address // _proxiedTokensManager
+    ) public virtual override(AgglayerBridgeL2) initializer {
+        revert InvalidInitializeFunction();
+    }
+
+    /**
+     * @dev initializer function to set the initial values when the contract is upgraded from the Etrog version
+     * @param _bridgeManager bridge manager address
+     * @param _emergencyBridgePauser emergency bridge pauser address, allowed to be zero if the chain wants to disable the feature to stop the bridge
+     * @param _emergencyBridgeUnpauser emergency bridge unpauser address, allowed to be zero if the chain wants to disable the feature to unpause the bridge
+     * @param _proxiedTokensManager address of the proxied tokens manager
+     */
+    function initializeBridgeZkEVM(
+        address _bridgeManager,
+        address _emergencyBridgePauser,
+        address _emergencyBridgeUnpauser,
+        address _proxiedTokensManager
+    ) public virtual reinitializer(3) {
```

Comment:
I think it's a good idea to add the `getInitializedVersion` as it is done in the `AgglayerBridge` `initialize`; check initializer version is 1 and also check that current values to update are zero.